File: design-patterns/go/builder/vehicle/plane.go
// plane.go

package main

import "fmt"

type Plane struct {
	wheel    int
	engine   int
	seat     int
	door     int
	wing     int
	interior bool
}

func NewPlane(noOfWheel int, noOfEngine int, noOfSeat int, noOfDoor int, wing int, interior bool) (rcvr *Plane) {
	rcvr = &Plane{}
	rcvr.wheel = noOfWheel
	rcvr.engine = noOfEngine
	rcvr.seat = noOfSeat
	rcvr.door = noOfDoor
	rcvr.wing = wing
	rcvr.interior = interior
	return
}

func (rcvr *Plane) GetDoor() int {
	return rcvr.door
}

func (rcvr *Plane) GetEngine() int {
	return rcvr.engine
}

func (rcvr *Plane) GetSeat() int {
	return rcvr.seat
}

func (rcvr *Plane) GetWheel() int {
	return rcvr.wheel
}

func (rcvr *Plane) GetWing() int {
	return rcvr.wing
}

func (rcvr *Plane) IsInterior() bool {
	return rcvr.interior
}

func (rcvr *Plane) ToString() string {
	return fmt.Sprintf("%v%v%v%v%v%v%v%v%v%v%v%v", "Plane: Wheel -> ", rcvr.wheel, " | Engine -> ", rcvr.engine, " | Seat -> ", rcvr.seat, " | Door -> ", rcvr.door, " | Wing: ", rcvr.wing, " | Interior -> ", rcvr.interior)
}

File: design-patterns/ts/builder/vehicle/plane.ts
// plane.ts

class Plane {
    private wheel: number;
    private engine: number;
    private seat: number;
    private door: number;
    private wing: number;
    private interior: boolean;

    constructor(
        noOfWheel: number,
        noOfEngine: number,
        noOfSeat: number,
        noOfDoor: number,
        wing: number,
        interior: boolean) {
        this.wheel = noOfWheel;
        this.engine = noOfEngine;
        this.seat = noOfSeat;
        this.door = noOfDoor;
        this.wing = wing;
        this.interior = interior;
    }

    getWheel(): number {
        return this.wheel;
    }

    getEngine(): number {
        return this.engine;
    }

    getSeat(): number {
        return this.seat;
    }

    getDoor(): number {
        return this.door;
    }

    getWing(): number {
        return this.wing;
    }

    isInterior(): boolean {
        return this.interior;
    }

    toString(): string {
        return `Plane:
        Wheel -> ${this.wheel}
        Engine -> ${this.engine}
        Seat -> ${this.seat}
        Door -> ${this.door}
        Wing -> ${this.wing}
        Interior -> ${this.interior}`;
    }
}

export default Plane;